src/ipv6/ndpv6.ml

```diff
@@ -665,6 +673,60 @@ module NeighborCache = struct
     | Not_found -> false
 end
 
+module RDNSSList = struct
+
+  type t =
+    (Ipaddr.t * time) list
+
+  let empty =
+    []
+
+  let to_list rdnssl =
+    List.map fst rdnssl
+
+  let add rdnssl ~now ?(lifetime = Duration.of_year 1) ip =
```

Comment:
From RFC 8106:
The value of Lifetime SHOULD by default be at least 3 * MaxRtrAdvInterval

Can we use that as default, and not 1 year :)

Reply:
I'm not sure how to access MaxRtrAdvInterval here as it's something that's set by the router.

Reply:
RFC 4861 says about MaxRtrAdvInterval:

> MUST be no less than 4 seconds and no greater than 1800 seconds.

So perhaps 1½ hours is reasonable.

Reading the code and some parts of RFC 8106 it seems to me that the lifetime is only `None` when it's zero in the packet, and in the rfc it says

> A value  of zero means that the RDNSS addresses MUST no longer be used

So maybe we should either promote `lifetime` to a regular labeled argument and optionally assert lifetime is greater than zero, or we should not have a default value and do nothing if `lifetime` is `None`.

I wonder if something similar applies to `RouterList.add` where there's a FIXME comment about the default lifetime...

Reply:
... or perhaps instead of doing nothing we should actively remove that IP address from the list.